App.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+
+import { Provider } from 'react-redux';
+import Store from "./store/store.js";
+
+import { PersistGate } from 'redux-persist/integration/react';
+import { Persistor } from './store/store.js';
+
+import { createStackNavigator } from '@react-navigation/stack';
+import { NavigationContainer } from "@react-navigation/native"
+
+import Icon from "react-native-vector-icons/FontAwesome";
+import { useNavigation } from '@react-navigation/native';
+
+import StartScreen from './components/screens/StartScreen';
+import LoginScreen from './components/screens/LoginScreen.js';
+import RegisterScreen from './components/screens/RegisterScreen.js';
+import Dashboard from './components/screens/Dashboard.js';
+import CustomButton from './components/reusable/CustomButton.js';
+
+const Stack = createStackNavigator();
+
+const StackNavigator = () => {
+  const navigation = useNavigation()
+  return (
+    <Stack.Navigator
+      initialRouteName='StartScreen'
+      screenOptions={{
+        headerShown: false,
+      }}
+    >
+      <Stack.Screen
+        name='StartScreen'
+        component={StartScreen}
+      />
+      <Stack.Screen
+        name='LoginScreen'
+        component={LoginScreen}
+      />
+      <Stack.Screen
+        name='RegisterScreen'
+        component={RegisterScreen}
+        options={{
+          headerShown: true,
+          headerTitle: "",
+          headerLeft: () =>
+            <View
+              style={{
```

Comment:
please create component for this and show it when the canGoBack is true, you can get the vanBoNack function from the navigation I think it's props for the header left, the component should be in the screenOptions for the navigator, and use goBack instead of navigate